### PR TITLE
returning 404 when no file in repo on /:name/-/:file

### LINKED ISSUE #41: Server crash on '/:name/-/:file'
The server crashes on '/:name/-/:file' requests, such as requesting this dependency:
http:/myserver:8082/grunt-cli/-/grunt-cli-0.1.13.tgz

```
Starting to load packages in repo_content/packages
Done auto-loading packages
Registered package delegator@0.0.1

http.js:691
    throw new Error('Can\'t set headers after they are sent.');
          ^
Error: Can't set headers after they are sent.
    at ServerResponse.OutgoingMessage.setHeader (http.js:691:11)
    at ServerResponse.format (/iad/common/apps/reggie/node_modules/reggie/node_modules/restify/lib/response.js:84:22)
    at ServerResponse.send (/iad/common/apps/reggie/node_modules/reggie/node_modules/restify/lib/response.js:197:32)
    at Server.<anonymous> (/iad/common/apps/reggie/node_modules/reggie/node_modules/restify/lib/index.js:87:21)
    at Server.EventEmitter.emit (events.js:106:17)
    at Domain.onError (/iad/common/apps/reggie/node_modules/reggie/node_modules/restify/lib/server.js:757:30)
    at Domain.EventEmitter.emit (events.js:95:17)
    at ReadStream.EventEmitter.emit (events.js:70:21)
    at fs.js:1500:12
    at /iad/common/apps/reggie/node_modules/reggie/node_modules/rimraf/node_modules/graceful-fs/graceful-fs.js:53:5
```

### PR DESCRIPTION
**Pull request by leftieFriele**
This fixes #41 were the reggie-server crashes when receiving a request for a file which doesn't exist in the local repo.
I made it return a 404 status code which I think makes sense.

### CODE REVIEW

**Comment by bajtos** [comment]
Hi @leftieFriele, thank you for the pull request. Could you please add an automated test? See [test/npm-protocol.js](https://github.com/mbrevoort/node-reggie/blob/master/test/npm-protocol.js).

**Comment by leftieFriele** [comment]
I would, except I find it really hard to understand how to actually prove the bug in a test given the setup in npm-protocol.js as I am not that familiar with how the npm client works.
I know what URL to pass into reggie-server in order to get it to crash, but I am not sure how to reproduce in the test harness. 
The kind of package declaration which fails is one such as requirejs which has the following: 

```
"_from": "requirejs@~2.1",
  "_resolved": "https://registry.npmjs.org/requirejs/-/requirejs-2.1.11.tgz"
```

**Comment by bajtos** [comment]
> I would, except I find it really hard to understand how to actually prove the bug in a test given the setup in npm-protocol.js as I am not that familiar with how the npm client works.

I see. In that case, can you write the test using [supertest](https://github.com/visionmedia/supertest) and simply check that `/requirejs/-/requirejs-2.1.11.tgz` returns 404 as expected? Something along these lines:

```
request(helpers.reggieUrl)
  .get('/requirejs/-/requirejs-2.1.11.tgz')
  .expect(404, done);
```

Since the test is very different from the rest of the tests in `npm-protocol.js`, it should go to a new file.

**Comment by leftieFriele** [comment]
I'll get on it first thing tomorrow, thanks for the tip on supertest. I've only used vows before.

**Comment by leftieFriele** [comment]
I've added the test-case for the bug and thanks again for the pointer on how to test this.

File: package.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "author": "Mike Brevoort <mike@brevoort.com>",
   "name": "reggie",
   "description": "An experimental light weight alternative to a full blown npm registry",
-  "version": "0.2.0",
+  "version": "0.2.0-FINN",
```
Comment (bajtos) [inline package.json]: Unintentional change? Please revert.

**Comment by bajtos** [comment]
@leftieFriele The test looks good, could you please change indentation to 2 spaces for the sake of consistency with other source files? Also add a newline at the end of the new file.

**Comment by leftieFriele** [comment]
oh that was by no means intentional, I've corrected the mistake and fixed indenting. sorry for wasting your time by being sloppy.

**Comment by bajtos** [comment]
I am afraid the latest commit made it only worse - check the diff here on github ([link](https://github.com/mbrevoort/node-reggie/pull/42/files)).

**Comment by leftieFriele** [comment]
finally it is fixed :+1:

**Comment by bajtos** [comment]
Now you are missing supertest dependency and the unit-test file is using like 6 spaces for indentation.

I have fixed those issues myself and landed your patch as 25ef94cb7. Thank you for the contribution!

**Comment by bajtos** [comment]
Released as v0.2.1

**Pull request closed without merging**